### PR TITLE
More library web ui

### PR DESCRIPTION
**Pull request by yipingliao**
@andrewconner @ahsu1230 cc @2is10 

Web library work for the day.
@ahsu1230  Thanks for backend support for published library sharing!
@andrewconner  I will ask for help re Merriweather CORS fun tomorrow.

Will work on keeping to library tonight/tomorrow AM before demo.

Yiping

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/common/services/angular-facebook-api.js [hunk 1]
```diff
@@ -123,16 +124,13 @@ angular.module('jun.facebook', [])
...
-				js = d.createElement(s);
-				js.id = id;
-				js.src = '//connect.facebook.net/en_US/all.js';
-				fjs.parentNode.insertBefore(js, fjs);
-			}(window.document, 'script', 'facebook-jssdk'));
+			(function(d, s, id){
+	        	var js, fjs = d.getElementsByTagName(s)[0];
+	        	if (d.getElementById(id)) {return;}
+	        	js = d.createElement(s); js.id = id;
+	        	js.src = '//connect.facebook.net/en_US/sdk.js';
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/common/services/angular-facebook-api.js]: Updated to latest and greatest FB sdk.

File: kifi-backend/modules/shoebox/angular/dev.html [hunk 2]
```diff
@@ -35,3 +35,4 @@
 <script src="/dist/lib.js"></script>
 <script src="/dist/kifi.js"></script>
 <script>angular.bootstrap(document, ['kifi']);</script>
+<script type="text/javascript" async src="//platform.twitter.com/widgets.js"></script>
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/dev.html:38]: Any better way of loading this Twitter code? @andrewconner told me that FB code is loaded lazily. I probably need to look into that. Will add a TODO.

**Comment by yipingliao** [comment]
On second thought, reverted the update to latest FB sdk. We don't need it for FB share, and I don't want to do any API updates w/o thoroughly tested all the FB-related features.

File: kifi-backend/modules/shoebox/angular/dev.html (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/dev.html:38]: Yep, I can show you how FB does it.

File: kifi-backend/modules/shoebox/angular/src/libraries/libraryCard.tpl.html [hunk 3]
```diff
@@ -53,21 +60,25 @@
...
+      {{editKeepsText}}
+    </button>
+    <button ng-if="canFollowLibrary(library)" class="kf-keep-lib-footer-button kf-keep-lib-footer-button-right kf-keep-lib-footer-button-follow-unfollow kf-keep-lib-footer-button-follow" ng-click="followLibrary(library)">Follow</button>
+    <button ng-if="alreadyFollowingLibrary(library)" class="kf-keep-lib-footer-button kf-keep-lib-footer-button-right kf-keep-lib-footer-button-follow-unfollow kf-keep-lib-footer-button-unfollow" ng-click="unfollowLibrary(library)"></button>
 
     <div kf-library-share-search ng-if="canBeShared(library)" library="library"></div>
     <button ng-if="canBeShared(library)" class="kf-keep-lib-footer-button">
       <span class="kf-keep-lib-footer-button-icon svg-twitter"></span>
-      <a href="https://twitter.com/share?url={{library.shareUrl}}" target="_blank">Share</a>
+      <a href="https://twitter.com/intent/tweet?original_referer={{library.shareUrl}}&text={{library.shareText}}&tw_p=tweetbutton&url={{library.shareUrl}}">Share</a>
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryCard.tpl.html:77]: Is we're just using this, do we need the twitter widget JS library at all?

**Comment by andrewconner** [comment]
Merging to get it out :)

File: kifi-backend/modules/shoebox/angular/dev.html (hunk 2 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/dev.html:38]: I’ll be happy to help figure out the lazy loading too. Let’s make it a priority.

**Comment by 2is10** [comment]
nice work

File: kifi-backend/modules/shoebox/angular/src/libraries/libraryCard.tpl.html (hunk 3 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryCard.tpl.html:77]: We won't need it if we just want to open the link in a new tab/window, but we need it to get the nice little modal - that's in the twitter js file.

**Pull request merged**